### PR TITLE
Add docs labels to workflow.md

### PR DESCRIPTION
**Pull request by AnotherButler**
Add labels the documentation repository uses.

### CODE REVIEW

File: docs/contributing/guidelines/workflow.md [hunk 1]
```diff
@@ -277,3 +277,12 @@ The following labels summarize the scope of the pull request:
 - *scope: feature*.
 - *scope: new-target*.
 - *scope: refactor*.
+
+### Documentation contribution labels
+
+The [documentation repository](https://github.com/ARMmbed/mbed-os-5-docs) uses some of the same labels that the `mbed-os` repository does. In addition, it includes: 
+
+- *code dependency* - The feature or change this pull request describes has not yet merged into `master` in `mbed-os`. Even if all the required reviewers have reviewed and approved this pull request, it must wait to mege until its code does.
```
Comment (iriark01) [inline docs/contributing/guidelines/workflow.md]:
```suggestion
- *code dependency* - The feature or change this pull request describes has not yet merged into `master` in `mbed-os`. Even if all the required reviewers have reviewed and approved this pull request, it must wait to merge until its code does.
```

**Comment by AnotherButler** [comment]
There are some labels in this repo we've never used. Shall we remove them?

File: docs/contributing/guidelines/workflow.md [hunk 2]
```diff
@@ -277,3 +277,12 @@ The following labels summarize the scope of the pull request:
...
+- *wait to merge* - This pull request is still a work in progress and is not yet ready for full review.
```
Comment (iriark01) [inline docs/contributing/guidelines/workflow.md]: I would say either "still in progress" or "is a work in progress". I don't think it needs both variants.
Also, I didn't know wait to merge refers to the review status; I assumed it was reviewed but not yet ready to merge for a reason other than code dependency.

**Review by iriark01 — CHANGES_REQUESTED** [review]
Thank you!

**Comment by iriark01** [comment]
> There are some labels in this repo we've never used. Shall we remove them?

Yes, clean it up. Thanks.

**Review by iriark01 — APPROVED** [review]

File: docs/contributing/guidelines/workflow.md [hunk 3]
```diff
@@ -277,3 +277,12 @@ The following labels summarize the scope of the pull request:
...
 - *scope: new-target*.
 - *scope: refactor*.
+
+### Documentation contribution labels
+
+The [documentation repository](https://github.com/ARMmbed/mbed-os-5-docs) uses some of the same labels that the `mbed-os` repository does. In addition, it includes: 
+
+- *code dependency* - The feature or change this pull request describes has not yet merged into `master` in `mbed-os`. Even if all the required reviewers have reviewed and approved this pull request, it must wait to merge until its code does.
+- *wait to merge* - This pull request is a work in progress and is not yet ready for full review. 
+- *Scripting work - not for writers* - This issue requires a change too large or frequent for us to do manually. Instead, a script needs to be created to fix this issue.
```
Comment (adbridge) [inline docs/contributing/guidelines/workflow.md:287]: My only comment here is that this is a very long label name ?

**Review by 0xc0170 — APPROVED** [review]

File: docs/contributing/guidelines/workflow.md (hunk 3 above)
Comment (AnotherButler) [reply docs/contributing/guidelines/workflow.md:287]: Would you recommend we change the label? If so, does how long the label has been like that matter?

**Pull request merged**